File: Backend/AstroGame.Shared/Models/Researches/DynamicResearchCost.cs
using AstroGame.Shared.Models.Technologies;

namespace AstroGame.Shared.Models.Researches
{
    public class DynamicResearchCost : TechnologyCost
    {
        /// <summary>
        /// The base value for the calculation
        /// </summary>
        public double BaseValue { get; set; }

        /// <summary>
        /// The multiplier for the calculation
        /// </summary>
        public double Multiplier { get; set; }

        public virtual LevelableResearch LevelableResearch { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/researches/dynamic_research_cost.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/resources/dynamic_cost.dart';
import 'package:astrogame_app/models/technologies/technology_cost.dart';
import 'package:json_annotation/json_annotation.dart';

part 'dynamic_research_cost.g.dart';

@GuidConverter()
@JsonSerializable()
class DynamicResearchCost extends TechnologyCost with DynamicCost {
  late double baseValue;
  late double multiplier;

  DynamicResearchCost();

  factory DynamicResearchCost.fromJson(Map<String, dynamic> json) =>
      _$DynamicResearchCostFromJson(json);
  Map<String, dynamic> toJson() => _$DynamicResearchCostToJson(this);
}
